kernel/mem_domain.c

```diff
@@ -104,20 +104,6 @@ void k_mem_domain_init(struct k_mem_domain *domain, uint8_t num_parts,
 
 	domain->num_partitions = 0U;
 	(void)memset(domain->partitions, 0, sizeof(domain->partitions));
-
-	if (num_parts != 0U) {
-		uint32_t i;
-
-		for (i = 0U; i < num_parts; i++) {
-			__ASSERT(check_add_partition(domain, parts[i]),
-				 "invalid partition index %d (%p)",
-				 i, parts[i]);
-
-			domain->partitions[i] = *parts[i];
-			domain->num_partitions++;
-		}
-	}
-
 	sys_dlist_init(&domain->mem_domain_q);
 
 #ifdef CONFIG_ARCH_MEM_DOMAIN_DATA
```

Comment:
just wondering @andrewboie , where is this option selected, and where is arch_mem_domain_init() implemented? Is this supposed to be out-of-tree?

Reply:
forthcoming paging work.